### ISSUE TITLE
RouterConfig data isn't properly chained on WithDeploy in certain scenarios

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#546]
This is a bug inside `Props` - essentially our `WithDeploy` code should read like this:

    def withDeploy(d: Deploy): Props = copy(deploy = d withFallback deploy)

From the original Scala source: https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/actor/Props.scala#L233

Our current `WithDeploy` code doesn't attempt to append a Fallback at all, which causes a perfectly valid configuration like this:

````
var router = masterActorSystem.ActorOf(new RoundRobinPool(2).Props(Props.Create<Echo>())
                .WithDeploy(
                new Deploy(new RemoteScope(intendedRemoteAddress.Copy()))), "remote-blub2");
````

To lose its `RouterConfig` during deployment.

**Comment by Aaronontheweb** [comment]
Note: when I attempt to change our `Props.WithDeploy` function to read like the above Scala function, the test runner barfs up an `AkkaConfiguratonException` on most tests ("cannot have self as a fallback.") 

So it's not a straightforward fix - in my next PR I have a hack inside Props that preserves `Deploy.RouterConfig` only. But really we should follow the Scala convention here and take the original `Deploy` as a fallback, rather than overwrite it.

**Comment by Aaronontheweb** [comment]
This is now resolved as of #707
